spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/jdbc/AmazonRdsDatabaseProperties.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.aws.autoconfigure.jdbc;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.style.ToStringCreator;
+
+/**
+ * Properties related to AWS Rds.
+ *
+ * @author Mete Alpaslan Katırcıoğlu
+ * @see org.springframework.cloud.aws.autoconfigure.jdbc.AmazonRdsDatabaseAutoConfiguration
+ */
+@ConfigurationProperties(prefix = "cloud.aws.rds")
+public class AmazonRdsDatabaseProperties {
+
+	/**
+	 * List of RdsInstances.
+	 */
+	private List<RdsInstance> instances = Collections.emptyList();
```

Comment:
I would suggest to use `Map<String, RdsInstance>` so we can keep property configuration `cloud.aws.rds.<instance-name>` and keep backward compatibility

Reply:
Actually, while adding configuration properties for RDS, the main reason of PR is lack of current RDS configuration and reshaping it to be more flexible as detailed in #gh-495

I'm not sure if we keep this backward compatible as it is. @eddumelendez

Reply:
I think in yaml, list looks much better than properties. My suggestion was due to we can see a the properties assigned to one instance as you can see below:

```
cloud.aws.rds.my-instance.username=

cloud.aws.rds.my-instance.password=
```

or

```
cloud:
  aws:
    rds:
      my-instance:
        username:
        password:
```

wdyt?

Reply:
Sorry @eddumelendez but I got a bit confused. I don't understand what is the differences between what you offer and what is written in https://cloud.spring.io/spring-cloud-aws/spring-cloud-aws.html#_configuring_data_source_in_spring_boot.

Instead, following configuration changes is requested in #495  

```
cloud:
  aws:
    rds:
      - instanceName: ${ENV_VARIABLE_FOR_INSTANCE_ONE}
        databaseName: database-name-one
        username: user-one
        password: secret-one
```

and the changes helps to define `instanceName` as a separate environment variable as below.

```
cloud:
  aws:
    rds:
	  instances:
	  	- instanceName: ${ENV_VARIABLE_FOR_INSTANCE_ONE}
	      databaseName: database-name-one
	      username: user-one
	      password: secret-one
```

Reply:
@mkatircioglu we should also consider @maciejwalkowiak's opinion here

Let me explain, using `Map<String, RdsInstance>` as suggested we can use the same semantic as before `cloud.aws.rds.<instance-name>`. What does it mean?

Having, for example the configuration properties below

```
cloud.aws.rds.my-instance-1.username=foo
cloud.aws.rds.my-instance-1.password=bar

cloud.aws.rds.my-instance-2.username=a
cloud.aws.rds.my-instance-2.password=b
```

Will create a map with 2 entries, one for `my-instance-1` and other for `my-instance-2` with the given `username` and `password`.

At first glance, looks like haven't change anything but the magic is done by the configurarion properties class that you are introducing

and it is only a suggestion :)

Reply:
I got your point @eddumelendez. 

Btw, are you also sure that dbInstanceIdentifier can be replaceable by an environment variable if we use `Map<String, RdsInstance>`in `AmazonRdsDatabaseProperties` as described in below?

```
cloud.aws.rds.${ENV_VARIABLE_FOR_INSTANCE_ONE}.username=foo
cloud.aws.rds.${ENV_VARIABLE_FOR_INSTANCE_ONE}.password=bar

cloud.aws.rds.${ENV_VARIABLE_FOR_INSTANCE_TWO}.username=a
cloud.aws.rds.${ENV_VARIABLE_FOR_INSTANCE_TWO}.password=b

ENV_VARIABLE_FOR_INSTANCE_ONE=my-instance-1
ENV_VARIABLE_FOR_INSTANCE_TWO=my-instance-2
```

If so, I can change as you suggested.

Btw, what is your opinion @maciejwalkowiak?

Reply:
AFAIR `cloud.aws.rds.${ENV_VARIABLE_FOR_INSTANCE_ONE}.username=foo` will not work and making it possible to externalize the instance id was the main motivation for this change.

Reply:
TBH, I am not sure about it :/

Reply:
According to your final comments, what is your latest decision? @eddumelendez @maciejwalkowiak

Reply:
I will go with the current (yours) implementation 👍🏽

I didn't take into account the environment variable resolution

Reply:
Great! So, there is no more review comments and I'm resolving this! Thanks a lot @eddumelendez